Frontend/ui-driver/src/Screens/DriverReferralScreen/View.purs

```diff
@@ -0,0 +1,341 @@
+    [ width MATCH_PARENT
+    , height WRAP_CONTENT
+    , orientation VERTICAL
+    , margin $ MarginTop 15
+    ][ textView $
+        [ width MATCH_PARENT
+        , height WRAP_CONTENT
+        , text $ getString RIDE_LEADERBOARD
+        , color Color.black800
+        ] <> FontStyle.body7 TypoGraphy
+
+     , linearLayout
+        [ width MATCH_PARENT
+        , height WRAP_CONTENT
+        , background state.data.config.primaryBackground
+        , cornerRadius 10.0
+        , padding $ Padding 15 15 15 15
+        , margin $ MarginTop 10
+        ][ imageView
+            [ width $ V 97
+            , height $ V 97
+            , imageWithFallback $ fetchImage FF_COMMON_ASSET "ny_driver_leaderboard"
+            ]
+          , linearLayout
+                [ width WRAP_CONTENT
+                , height WRAP_CONTENT
+                , orientation VERTICAL
+                , gravity CENTER
+                , margin $ MarginLeft 80
+                ][ textView $
+                    [ width MATCH_PARENT
+                    , height WRAP_CONTENT
+                    , text $ getString YOUR_RANK
+                    , color state.data.config.popupBackground
+                    , gravity CENTER
+                    ] <> FontStyle.body3 TypoGraphy
+                , textView $
+                    [ width MATCH_PARENT
+                    , height WRAP_CONTENT
+                    , text $ getString NOT_AVAILABLE_YET
+                    , color state.data.config.popupBackground
+                    , gravity CENTER
+                    , margin $ MarginTop 5
+                    ] <> FontStyle.body5 TypoGraphy
+                , textView
+                    [ width WRAP_CONTENT
+                    , height WRAP_CONTENT
+                    , text $ getString LEARN_MORE
+                    , background state.data.config.popupBackground
+                    , color state.data.config.primaryBackground
+                    , cornerRadius 10.0
+                    , padding $ Padding 25 8 25 8
+                    , gravity CENTER
+                    , onClick push $ const $ LearnMore
+                    , margin $ MarginTop 5
+                    ]
+                 ]
+              ]
+     ]
+
+bottomNavBarView :: forall w . (Action -> Effect Unit) -> DriverReferralScreenState -> PrestoDOM (Effect Unit) w
+bottomNavBarView push state =
+  linearLayout
+  [ width MATCH_PARENT
+  , height WRAP_CONTENT
+  ][BottomNavBar.view (push <<< BottomNavBarAction) (navData ScreenNames.REFERRAL_SCREEN state.data.config.bottomNavConfig)]
+
+appQRCodeView :: forall w . (Action -> Effect Unit) -> DriverReferralScreenState -> PrestoDOM (Effect Unit) w
+appQRCodeView push state = 
+  linearLayout
+    [ width MATCH_PARENT
+    , height MATCH_PARENT
+    , gravity CENTER
+    , background Color.blackLessTrans
+    ][
+      linearLayout
+        [ width MATCH_PARENT
+        , height WRAP_CONTENT
+        , orientation VERTICAL
+        , gravity CENTER
```

Comment:
instead of using `gravity CENTER` in every view try to use `layoutGravity "center"` in single parent view

Reply:
Since the orientation is vertical it is not working